dataactcore/scripts/populate_subaward_table.py

```diff
@@ -0,0 +1,147 @@
+        values = min_id
+    else:
+        operator = 'IN'
+        values = '({})'.format(','.join([str(id) for id in ids]))
+    sql = sql.format(operator, values)
+
+    # run the SQL
+    inserted = sess.execute(sql)
+    sess.commit()
+    inserted_count = inserted.rowcount
+    logger.info('Inserted {} sub-{} to the subaward table'.format(inserted_count, service_type))
+    return inserted_count
+
+
+def fix_broken_links(sess, service_type):
+    """ Attempts to resolve any unlinked subawards given the current data
+
+        Args:
+            sess: connection to the database
+            service_type: type of service to ping (usually 'procurement_service' or 'grant_service')
+
+        Raises:
+            Exception: service type is invalid
+    """
+    # get the broken links to delete later
+    subaward_type_map = {PROCUREMENT: 'sub-contract', GRANT: 'sub-grant'}
+    if service_type not in subaward_type_map:
+        raise Exception('Invalid service type provided: {}'.format(service_type))
+
+    sql = extract_subaward_sql(service_type, 'link')
+
+    # run the SQL
+    updated = sess.execute(sql)
+    sess.commit()
+
+    updated_count = updated.rowcount
+    logger.info('Updated {} sub-{} in the subaward table'.format(updated_count, service_type))
+    return updated_count
+
+
+if __name__ == '__main__':
+    now = datetime.datetime.now()
+    configure_logging()
+    parser = argparse.ArgumentParser(description='Pull data from FSRS Feed')
+    method = parser.add_mutually_exclusive_group(required=True)
+    parser.add_argument('-p', '--procurements', action='store_true', help="Load just procurement awards")
+    parser.add_argument('-g', '--grants', action='store_true', help="Load just grant awards")
+    method.add_argument('-b', '--backfill', action='store_true', help="Backfill")
+    method.add_argument('-i', '--ids', type=int, nargs='+',
+                        help="Single or list of FSRS ids to populate the subaward table")
+
+    with create_app().app_context():
+        logger.info("Begin backfilling Subaward table")
+        sess = GlobalDB.db().session
+        args = parser.parse_args()
+
+        metrics_json = {
+            'script_name': 'populate_subaward_table.py',
+            'records_inserted': 0,
+            'start_time': str(now)
+        }
+
+        service_types = []
+        if not (args.procurements or args.grants):
+            logger.error('FSRS types not provided. Please specify procurements, grants, or both.')
+            sys.exit(1)
+        if args.procurements:
+            service_types.append(PROCUREMENT)
+        if args.grants:
+            service_types.append(GRANT)
+
+        records_inserted = 0
+        for service_type in service_types:
+            if args.backfill:
+                records_inserted += populate_subaward_table(sess, service_type, min_id=0)
+            elif args.ids:
+                records_inserted += populate_subaward_table(sess, service_type, ids=args.ids)
```

Comment:
Is there any situation where this would need to be called with a min id other than 0? Or is the only time we do that when we call the function specifically and not the script overall?

Reply:
Good catch, replaced "backfill" with "min_id" to cover special cases.